tests/basic/pool_function_test.go

```diff
@@ -726,6 +726,82 @@ var _ = Describe("{PoolExpandWhileResizeDiskInProgress}", func() {
+
+    /*
+        Steps:
+            1. Initiate pool expand with resize-disk operation. 
+            2. Cycle the node through maintenance mode.
+            3. Verify pool expand operation goes to completion.
+    */
+
+    BeforeEach(func() {
+        StartTorpedoTest("MaintenanceCycleDuringPoolExpandResizeDisk",
+            "Perform maintenance cycle during pool expand with resize-disk operation", nil, testrailID)
+        contexts = scheduleApps()
+    })
+
+    JustBeforeEach(func() {
+        poolIDToResize = pickPoolToResize()
+        log.Infof("Picked pool %s to resize", poolIDToResize)
+        poolToResize = getStoragePool(poolIDToResize)
+        storageNode, err = GetNodeWithGivenPoolID(poolIDToResize)
+        log.FailOnError(err, "Failed to get node with given pool ID")
+    })
+
+    JustAfterEach(func() {
+        AfterEachTest(contexts)
+    })
+
+    AfterEach(func() {
+        appsValidateAndDestroy(contexts)
+        EndTorpedoTest()
+    })
+
+    stepLog := "cycle through maintenance mode during pool expand with resize-disk"
+    It(stepLog, func() {
+        log.InfoD(stepLog)
+
+        originalSizeInBytes = poolToResize.TotalSize
+        targetSizeInBytes = originalSizeInBytes + 100*units.GiB
+        targetSizeGiB = targetSizeInBytes / units.GiB
+
+        log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v GiB with type add-disk",
+            poolIDToResize, poolToResize.TotalSize/units.GiB, targetSizeGiB)
+        triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK)
+
+        // Enter Maintenance Mode
+        err = Inst().V.EnterMaintenance(*storageNode)
+        log.FailOnError(err, fmt.Sprintf("fail to enter node %s in maintenance mode", storageNode.Name))
+        status, err := Inst().V.GetNodeStatus(*storageNode)
+        log.FailOnError(err, fmt.Sprintf("Error getting PX status of node %s", storageNode.Name))
+        dash.VerifyFatal(*status, api.Status_STATUS_MAINTENANCE, fmt.Sprintf("Node %s Status not Online", storageNode.Name))
+
+        // Exit Maintenance Mode
+        err = Inst().V.ExitMaintenance(*storageNode)
+        log.FailOnError(err, fmt.Sprintf("fail to exit node %s in maintenance mode", storageNode.Name))
+        status, err = Inst().V.GetNodeStatus(*storageNode)
+        log.FailOnError(err, fmt.Sprintf("Error getting PX status of node %s", storageNode.Name))
+        dash.VerifyFatal(*status, api.Status_STATUS_OK, fmt.Sprintf("Node %s Status not Online", storageNode.Name))
+
+        err = waitForOngoingPoolExpansionToComplete(poolIDToResize)
+        dash.VerifyFatal(err, nil, "Pool expansion does not result in error")
+        verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
+
+        // verify pool size after maintenance cycle
+        verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
```

Comment:
looks duplicate here already you are verifying at 795

Reply:
Done. removed duplicate.